lldb/source/Utility/Status.cpp

```diff
@@ -174,33 +233,91 @@ const char *Status::AsCString(const char *default_error_str) const {
 
 // Clear the error and any cached error string that it might contain.
 void Status::Clear() {
-  m_code = 0;
-  m_type = eErrorTypeInvalid;
-  m_string.clear();
+  if (m_error)
+    LLDB_LOG_ERRORV(GetLog(LLDBLog::API), std::move(m_error),
+                    "dropping error {0}");
+  m_error = llvm::Error::success();
+  llvm::consumeError(std::move(m_error));
 }
 
 // Access the error value.
-Status::ValueType Status::GetError() const { return m_code; }
+Status::ValueType Status::GetError() const {
+  Status::ValueType result = 0;
+  llvm::visitErrors(m_error, [&](const llvm::ErrorInfoBase &error) {
+    std::error_code ec = error.convertToErrorCode();
+    if (ec.category() == std::generic_category() ||
+        ec.category() == generic_category() ||
+        ec.category() == expression_category())
+      result = ec.value();
```

Comment:
How about just returning `ec.value()` regardless of the category?

Reply:
?

Reply:
This special case only exists because it otherwise breaks the GDBRemoteCommunicationServer unit test. Probably not worth preserving, but I didn't want to unnecessarily change this behavior either.

Reply:
It's the SendErrorResponse_StringError test.

Reply:
I would actually prefer changing the code over there if it meant simplifying this. Would #108170 do the trick?

Reply:
I incorporate the change. There's still some amount of compatibility-wrapping needed, but it got better.